src/campaign/entity/Campaign.entity.ts

```diff
@@ -0,0 +1,85 @@
+import {
+    Entity,
+    PrimaryGeneratedColumn,
+    Column,
+    ManyToOne,
+    JoinColumn,
+} from 'typeorm';
+import { CampaignStatus } from '../status/campaign-status.enum';
+import { MailContentType } from 'src/template/types/Mail-content.type';
+import { Template } from 'src/template/entity/Template.entity';
+import { EmailProvider } from '../../email-provider/entity/Email-provider.entity';
+
+@Entity()
+export class Campaign {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column()
+    name: string;
+
+    @Column()
+    description: string;
+
+    @Column({
+        type: 'enum',
+        enum: CampaignStatus,
+        default: CampaignStatus.PREPARING,
+    })
+    status: CampaignStatus;
+
+    @ManyToOne(() => Template, (template) => template.campaigns, {
+        eager: true,
+    })
+    @JoinColumn()
+    template: Template;
+
+    @ManyToOne(
+        () => EmailProvider,
+        (eMailProvider) => eMailProvider.campaigns,
+        {
+            eager: true,
+        },
+    )
+    @JoinColumn()
+    eMailProvider: EmailProvider;
+
+    @Column('json', { nullable: true })
+    content: MailContentType;
```

Comment:
jak widzisz zeby to pole dzialalo?

Reply:
`export type MailContentType = {
    logoUrl: string;
    logoAlt: string;
    mainTitle: string;
    subtitle?: string;
    content: ContentType[];
    footerLongText: string;
    footerTextLink: string;
    footerLinkUrl: string;
    footerLogoUrl: string;
    footerLogoAlt: string;
};

type ContentType = {
    title: string;
    subtitle: string;
    text: string;
    linkText?: string;
    linkUrl?: string;
    imageUrl?: string;
    imageAlt?: string;
};`

Do realizacji kampani będzie musiała być dostarczona w jakiś sposób treść (content) do szablonu maila. Mój pomysł jest, że każdy szablon (templatka) będzie zawierała w sobie zmienne do zastąpienia treścią a zmienne te będą nazywały się tak jak klucze tego obiektu powyżej. Mamy więc narzucony kształt danych, ale niektóre klucze są opcjonalne, tablica z treścią główną też może mieć różna wielkość. W efekcie na każdym szablonie można umieszczać bardzo różną treść, dość elastycznie - od krótkich po bardzo długie mailingi. Apka będize zastępowała zmienne z templatki wartościami poszczególnych kluczy. Taki kontent w odpowiednim kształcie danych będzie trzymany właśnie tutaj.

Reply:
jest to jakis pomysl... zastanwiam sie nad wielkoscia tych pol - dodalbym ograniczenia co do wielkosci tego w bazie bo tak to moze byc ze ktos wam w kazdym polu da maksymalna wartosc dla varchar i bedzie problem

Reply:
ok, zrobię ograniczenie i będzie walidacja